app/src/main/java/org/kiwix/kiwixmobile/wifi_hotspot/WifiHotspotManager.java

```diff
@@ -0,0 +1,89 @@
+package org.kiwix.kiwixmobile.wifi_hotspot;
+
+import android.content.Context;
+import android.net.wifi.WifiConfiguration;
+import android.net.wifi.WifiManager;
+import android.os.Build;
+import android.os.Handler;
+import android.util.Log;
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+import org.kiwix.kiwixmobile.webserver.ServerStateListener;
+
+/**
+ * WifiHotstopManager class makes use of the Android's WifiManager and WifiConfiguration class
+ * to implement the wifi hotspot feature.
+ * Created by Adeel Zafar on 28/5/2019.
+ */
+
+public class WifiHotspotManager {
+  private final WifiManager wifiManager;
+  private final Context context;
+  WifiManager.LocalOnlyHotspotReservation hotspotReservation;
+  boolean isHotspotEnabled;
+  WifiConfiguration currentConfig;
+  private static final String TAG = "WifiHotspotManager";
+
+  public WifiHotspotManager(@NonNull Context context) {
+    this.context = context;
+    wifiManager = (WifiManager) this.context.getSystemService(Context.WIFI_SERVICE);
+  }
+
+  //Workaround to turn on hotspot for Oreo versions
+  @RequiresApi(api = Build.VERSION_CODES.O)
+  public void turnOnHotspot(@NonNull ServerStateListener serverStateListener) {
+    wifiManager.startLocalOnlyHotspot(new WifiManager.LocalOnlyHotspotCallback() {
+
+      @Override
+      public void onStarted(WifiManager.LocalOnlyHotspotReservation reservation) {
+        super.onStarted(reservation);
+        hotspotReservation = reservation;
+        currentConfig = hotspotReservation.getWifiConfiguration();
+
+        Log.v(TAG, "THE PASSWORD IS: "
```

Comment:
ehhh so many lines for a log statement. if you need this message extract a method `print(currentConfig)` so it doesn't interrupt reading what is happening

Reply:
Fixed. df0a5e6eb11791f38114646c735bdfcfd8c6df3c